src/cascade/model/operations.py

```diff
@@ -4,282 +4,198 @@
-    not_constant = complete_grid[no_const_value & not_squeezed]
-
-    # Iterate through the non-constant, and look at value priors.
-    mutable_value = not_constant[not_constant.prior_type == "value"]
-    var = dismod_file.var
-    for index, row in mutable_value.iterrows():
-        age_id = row["age_id"]
-        time_id = row["time_id"]
-        var_id = var[(var.smooth_id == smooth_id) & (var.age_id == age_id) & (var.time_id == time_id)].var_id.values[0]
-        sub_draws = draws[draws.fit_var_id == var_id]
-        value = sub_draws.fit_var_value.mean()
-        std = sub_draws.fit_var_value.std()
-        complete_grid.loc[(complete_grid.age_id == age_id) & (complete_grid.time_id == time_id), "mean"] = value
-        complete_grid.loc[(complete_grid.age_id == age_id) & (complete_grid.time_id == time_id), "std"] = std
-
-    # Change them back to being in the Dismod form.
-    smooth, reduced_smooth_grid, reduced_prior = reduce_priors(smooth, complete_grid)
-    return smooth, reduced_smooth_grid, reduced_prior
+    value_priors = smooth.value_priors
+    for row in estimate_at.itertuples():
+        # If we don't copy the prior, then we modify in-place, which changes
+        # the whole grid of priors by accident.
+        prior = copy(value_priors[row.age, row.time].prior)
+        assert prior is not None, f"none at ({row.age}, {row.time})"
+        is_constant = "lower" in dir(prior) and prior.lower >= prior.upper
+        is_constant |= isinstance(prior, Constant)
+        if is_constant:
+            continue
+
+        std_ok = row.std > MINIMUM_STANDARD_DEVIATION_ABSOLUTE
+        std_ok |= row.mean > 0 and row.std / row.mean > MINIMUM_STANDARD_DEVIATION_RELATIVE
+
+        if std_ok:
+            if hasattr(prior, "mean"):
+                prior.mean = row.mean
+            if hasattr(prior, "standard_deviation"):
+                prior.standard_deviation = row.std
+            value_priors[row.age, row.time].prior = prior
+        else:
+            value_priors[row.age, row.time].prior = Constant(prior.mean)
+
+
+def _covariate_name_to_smooth(covariate_name, local_covariates, mulcovs):
+    """Find in this model context the Smooth for a given covariate name."""
+    covariate_objs = [cobj for cobj in local_covariates if cobj.name == covariate_name]
+    if not covariate_objs:
+        return None
+    match_mulcov = [mc for mc in mulcovs if mc.column == covariate_objs[0]]
+    if not match_mulcov:
+        return None
+    return match_mulcov[0].smooth
+
+
+def _estimates_from_one_grid(field_df):
+    """Given a dataframe with all var draws for a single field, return
+    one var table with mean and standard deviation."""
+    # Exclude mulstd to get just the grid values.
+    exclude_mulstd = field_df[~field_df.var_type.str.startswith("mulstd")]
+    grid_df = exclude_mulstd.set_index("fit_var_id")
+    # Gives grid with one age-time for each var_id.
+    with_at = grid_df[["age", "time"]].groupby(level=0).mean()
```

Comment:
This groups by the index? Ok. I don't use that pattern so I have to sit and think about it. I'd probably not set the index and explicitly group by "fit_var_id". Not saying that's better, just what I'm used to.

Reply:
It groups by the index. I chose this because it enables the `join` command later, which requires aligned indices.

Reply:
Yeah, that makes sense. Good.